src/main/java/org/wildfly/security/auth/realm/ldap/LdapSecurityRealm.java

```diff
@@ -147,17 +153,29 @@ public void registerIdentityChangeListener(Consumer<Principal> listener) {
         DirContext dirContext = null;
         try {
             dirContext = obtainContext();
-            EventContext eventContext = (EventContext) dirContext.lookup("");
-            eventContext.addNamingListener("", EventContext.SUBTREE_SCOPE, new ServerNotificationListener(listener));
+            innerRegisterIdentityChangeListener(dirContext,listener);
         } catch (Exception cause) {
-            throw log.ldapRealmFailedRegisterListener(cause);
+            //either connection died or realm not available during boot
+            synchronized (this.listenersPendingRegistration) {
+                //we need to wait, lets store
+                listenersPendingRegistration.add(listener);
+            }
+            log.ldapRealmDeferRegistration(this.name);
+            if(log.isDebugEnabled()){
+                log.debug("Listener registration failure: ", cause);
+            }
         } finally {
             if (dirContext != null) {
                 closeContext(dirContext);
             }
         }
     }
 
+    private void innerRegisterIdentityChangeListener(final DirContext dirContext, final Consumer<Principal> listener) throws NamingException {
```

Comment:
well, I just was oblivious to fact that both methods have different set of params. ech...